src/core/coap/coap_secure.cpp

```diff
@@ -55,8 +55,28 @@ CoapSecure::CoapSecure(Instance &aInstance)
     , mTransmitMessage(NULL)
     , mTransmitTask(aInstance, &CoapSecure::HandleUdpTransmit, this)
 {
+    mLayerTwoSecurity      = false;
+    mApplicationCoapSecure = false;
 }
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+CoapSecure::CoapSecure(Instance &       aInstance,
+                       Tasklet::Handler aUdpTransmitHandle,
+                       Timer::Handler   aRetransmissionTimer,
+                       Timer::Handler   aResponsesQueueTimer)
+    : CoapBase(aInstance, aRetransmissionTimer, aResponsesQueueTimer)
+    , mConnectedCallback(NULL)
+    , mConnectedContext(NULL)
+    , mTransportCallback(NULL)
+    , mTransportContext(NULL)
+    , mTransmitMessage(NULL)
+    , mTransmitTask(aInstance, aUdpTransmitHandle, this)
+{
+    mLayerTwoSecurity      = true;
```

Comment:
nit: prefer initializer list

Reply:
done

Reply:
Can initialization of `mLayerTwoSecurity` and `mApplicationCoapSecure` be moved into the initializer list?

Reply:
Yes. Done.